components/launcher/src/sys/unix/service.rs

```diff
@@ -85,33 +75,12 @@ impl Process {
         }
     }
 
-    pub fn try_wait(&mut self) -> Result<Option<ExitStatus>> {
-        if let Some(status) = self.status {
-            return Ok(Some(status));
-        }
-        let mut status = 0 as c_int;
-        match unsafe { libc::waitpid(self.pid, &mut status, libc::WNOHANG) } {
-            0 => Ok(None),
-            -1 => Err(Error::ExecWait(io::Error::last_os_error())),
-            _ => {
-                self.status = Some(ExitStatus::from_raw(status));
-                Ok(Some(ExitStatus::from_raw(status)))
-            }
-        }
+    pub fn try_wait(&mut self) -> io::Result<Option<ExitStatus>> {
+        self.0.try_wait()
```

Comment:
As you can see [the stdlib implementation is nearly identical](https://github.com/rust-lang/rust/blob/b7c6e8f1805cd8a4b0a1c1f22f17a89e9e2cea23/src/libstd/sys/unix/process/process_unix.rs#L390).